src/zcl_abapgit_repo.clas.abap

```diff
@@ -107,37 +108,55 @@ CLASS zcl_abapgit_repo DEFINITION
         VALUE(rt_list) TYPE scit_alvlist
       RAISING
         zcx_abapgit_exception .
-
+    METHODS has_remote_source
+    ABSTRACT
+      RETURNING
+        VALUE(rv_yes) TYPE abap_bool .
+    METHODS status
+      IMPORTING
+        !io_log TYPE REF TO zcl_abapgit_log OPTIONAL
+      RETURNING
+        VALUE(rt_results) TYPE zif_abapgit_definitions=>ty_results_tt
+      RAISING
+        zcx_abapgit_exception .
   PROTECTED SECTION.
 
+    INTERFACE zif_abapgit_definitions LOAD .
+    TYPE-POOLS abap .
     DATA mt_local TYPE zif_abapgit_definitions=>ty_files_item_tt .
     DATA mt_remote TYPE zif_abapgit_definitions=>ty_files_tt .
-    DATA mv_do_local_refresh TYPE abap_bool .
+    DATA mv_request_local_refresh TYPE abap_bool .
     DATA mv_last_serialization TYPE timestamp .
     DATA ms_data TYPE zif_abapgit_persistence=>ty_repo .
     DATA mv_code_inspector_successful TYPE abap_bool .
+    DATA mv_remote_is_initialized TYPE abap_bool .
+    DATA mt_status TYPE zif_abapgit_definitions=>ty_results_tt .
 
     METHODS set
-      IMPORTING it_checksums       TYPE zif_abapgit_persistence=>ty_local_checksum_tt OPTIONAL
-                iv_url             TYPE zif_abapgit_persistence=>ty_repo-url OPTIONAL
-                iv_branch_name     TYPE zif_abapgit_persistence=>ty_repo-branch_name OPTIONAL
-                iv_head_branch     TYPE zif_abapgit_persistence=>ty_repo-head_branch OPTIONAL
-                iv_offline         TYPE zif_abapgit_persistence=>ty_repo-offline OPTIONAL
-                is_dot_abapgit     TYPE zif_abapgit_persistence=>ty_repo-dot_abapgit OPTIONAL
-                is_local_settings  TYPE zif_abapgit_persistence=>ty_repo-local_settings OPTIONAL
-                iv_deserialized_at TYPE zif_abapgit_persistence=>ty_repo-deserialized_at OPTIONAL
-                iv_deserialized_by TYPE zif_abapgit_persistence=>ty_repo-deserialized_by OPTIONAL
-      RAISING   zcx_abapgit_exception .
-
+      IMPORTING
+        !it_checksums       TYPE zif_abapgit_persistence=>ty_local_checksum_tt OPTIONAL
+        !iv_url             TYPE zif_abapgit_persistence=>ty_repo-url OPTIONAL
+        !iv_branch_name     TYPE zif_abapgit_persistence=>ty_repo-branch_name OPTIONAL
+        !iv_head_branch     TYPE zif_abapgit_persistence=>ty_repo-head_branch OPTIONAL
+        !iv_offline         TYPE zif_abapgit_persistence=>ty_repo-offline OPTIONAL
+        !is_dot_abapgit     TYPE zif_abapgit_persistence=>ty_repo-dot_abapgit OPTIONAL
+        !is_local_settings  TYPE zif_abapgit_persistence=>ty_repo-local_settings OPTIONAL
+        !iv_deserialized_at TYPE zif_abapgit_persistence=>ty_repo-deserialized_at OPTIONAL
+        !iv_deserialized_by TYPE zif_abapgit_persistence=>ty_repo-deserialized_by OPTIONAL
+      RAISING
+        zcx_abapgit_exception .
```

Comment:
suggest when making changes to core functionallity, to split the formatting into a separate commit and/or PR, it would have made it easier to review

Reply:
Agree. reverted to the original.
Although this was a standard pretty print. A bit difficult to isolate them as it is done by se80.

BTW is there any description about recommended editor settings to contribute to AG ? Sometimes I edit code in se80, sometimes in eclipse. And automatic replacements are quite annoying. I think it could be nice to have some "standard recommendation" somewhere in contribution.md ...

Reply:
I use SE80, but I dont think we can make it the standard, and I'd like not to require Eclipse. Over time, files will have SE80 standard